modules/context/captcha.go

```diff
@@ -73,6 +76,14 @@ func VerifyCaptcha(ctx *Context, tpl base.TplName, form interface{}) {
 		valid, err = hcaptcha.Verify(ctx, ctx.Req.Form.Get(hCaptchaResponseField))
 	case setting.MCaptcha:
 		valid, err = mcaptcha.Verify(ctx, ctx.Req.Form.Get(mCaptchaResponseField))
+	case setting.CfTurnstile:
+		var ip string
+		if setting.Service.CfReverseProxyHeader == "" {
+			ip = ctx.RemoteAddr()
+		} else {
+			ip = ctx.Req.Header.Get(setting.Service.CfReverseProxyHeader)
+		}
```

Comment:
I don't think we need this,  it's OK to just use `ip := ctx.RemoveAddr()`

The middleware will set the remote addr to real ip according to configuration:

https://github.com/go-gitea/gitea/blob/9f919cf08339e8ec8f6b0b2a0dcafb3ea905fbec/routers/common/middleware.go#L36-L48

So the setting `CfReverseProxyHeader` is also unnecessary.

Reply:
BTW, the `ctx.RemoteAddr()` returns "ip:port", not ip, I'm not sure if this is OK to cloudflare turnstile.

Reply:
This was an oversight on my part, what cloudflare needs is the user's ip address. `ip := ctx.RemoveAddr()` can get user's real ip then I will delete `CfReverseProxyHeader`.

Reply:
I have a question, the ip obtained by `ctx.RemoteAddr()` may be either ipv4 or ipv6. Is this the case?

Reply:
> I have a question, the ip obtained by `ctx.RemoteAddr()` may be either ipv4 or ipv6. Is this the case?

That's right.